UefiCpuPkg/Library/MpInitLib/MpLib.c

```diff
@@ -2152,7 +2154,14 @@ MpInitLibInitialize (
   // Duplicate BSP's IDT to APs.
   // All APs share one separate IDT. So AP can get the address of CpuMpData by using IDTR.BASE + IDTR.LIMIT + 1
   //
-  CopyMem ((VOID *)ApIdtBase, (VOID *)VolatileRegisters.Idtr.Base, VolatileRegisters.Idtr.Limit + 1);
+  // Maintain compatibility with PeiServicesTablePointerLibIdt which stores the *PeiServices at
+  // Idt.Base - sizeof(32 bit pointer);

```

Comment:
Why 32 bit pointer is assumed? 
I see pure 64 bit platform is using PeiServicesTablePointerLibIdt.

For example 
https://github.com/tianocore/edk2/blob/master/OvmfPkg/AmdSev/AmdSevX64.dsc#L237

May be it is required to avoid UINTN and use UINT64 at
https://github.com/tianocore/edk2/blob/master/MdePkg/Library/PeiServicesTablePointerLibIdt/PeiServicesTablePointer.c#L41 ?

Reply:
@ghbaccount, I agree, I actually have made the change to be the same as PeiServicesTablePointer.c and use UINTN, just about to push that up. When this change was originally made there was not 64 bit PEI